server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequest.java

```diff
@@ -114,178 +130,65 @@ public NodesStatsRequest indices(boolean indices) {
-        return Metric.FS.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node file system stats be returned.
-     */
-    public NodesStatsRequest fs(boolean fs) {
-        addOrRemoveMetric(fs, Metric.FS.metricName());
-        return this;
+    public Set<String> requestedMetrics() {
+        return Set.copyOf(requestedMetrics);
     }
 
     /**
-     * Should the node Transport be returned.
+     * Add metric
      */
-    public boolean transport() {
-        return Metric.TRANSPORT.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node Transport be returned.
-     */
-    public NodesStatsRequest transport(boolean transport) {
-        addOrRemoveMetric(transport, Metric.TRANSPORT.metricName());
-        return this;
-    }
-
-    /**
-     * Should the node HTTP be returned.
-     */
-    public boolean http() {
-        return Metric.HTTP.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node HTTP be returned.
-     */
-    public NodesStatsRequest http(boolean http) {
-        addOrRemoveMetric(http, Metric.HTTP.metricName());
-        return this;
-    }
-
-    public boolean breaker() {
-        return Metric.BREAKER.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node's circuit breaker stats be returned.
-     */
-    public NodesStatsRequest breaker(boolean breaker) {
-        addOrRemoveMetric(breaker, Metric.BREAKER.metricName());
-        return this;
-    }
-
-    public boolean script() {
-        return Metric.SCRIPT.containedIn(requestedMetrics);
-    }
-
-    public NodesStatsRequest script(boolean script) {
-        addOrRemoveMetric(script, Metric.SCRIPT.metricName());
-        return this;
-    }
-
-
-    public boolean discovery() {
-        return Metric.DISCOVERY.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node's discovery stats be returned.
-     */
-    public NodesStatsRequest discovery(boolean discovery) {
-        addOrRemoveMetric(discovery, Metric.DISCOVERY.metricName());
+    public NodesStatsRequest addMetric(String metric) {
+        if (Metric.allMetrics().contains(metric) == false) {
+            throw new IllegalStateException("Used an illegal metric: " + metric);
+        }
+        requestedMetrics.add(metric);
         return this;
     }
 
-    public boolean ingest() {
-        return Metric.INGEST.containedIn(requestedMetrics);
-    }
-
     /**
-     * Should ingest statistics be returned.
+     * Add an array of metric names
      */
-    public NodesStatsRequest ingest(boolean ingest) {
-        addOrRemoveMetric(ingest, Metric.INGEST.metricName());
+    public NodesStatsRequest addMetrics(String... metrics) {
+        // use sorted set for reliable ordering in error messages
+        SortedSet<String> illegalMetrics = new TreeSet<>();
+        Set<String> validMetrics = new HashSet<>();
+        for (String metric : metrics) {
+            if (Metric.allMetrics().contains(metric)) {
```

Comment:
Could we simplify this by using `Set.containsAll`? Essentially this:
```
Set<String> metricsSet = new TreeSet<>(metrics);
if (Metric.allMetrics().containsAll(metricSet) == false) {
    Set<String> illegalMetrics = metricsSet.removaAll(Metric.allMetrics());
    // throw
}
requestMetrics.addAll(metricsSet);
```

Reply:
Ah, that looks better. I'll do this.